src/scheduler/job_info.c

```diff
@@ -3252,12 +3252,9 @@ find_jobs_to_preempt(status *policy, resource_resv *hjob, server_info *sinfo, in
 
 		update_universe_on_end(npolicy, pjob,  "S", NO_ALLPART);
 		rjobs_count--;
-		if (nsinfo->calendar != NULL) {
-			te = find_timed_event(nsinfo->calendar->events, 0, pjob->name, TIMED_END_EVENT, 0);
-			if (te != NULL) {
-				if (delete_event(nsinfo, te, DE_NO_FLAGS) == 0)
+		if (pjob->end_event != NULL) {
+				if (delete_event(nsinfo, pjob->end_event, DE_NO_FLAGS) == 0)
```

Comment:
There are two cases where we call update_server_on_end().  The first is here, and the other is in calendaring.  In calendaring, we process and move past each event as we call update_universe_on_end().  We then don't care about that event again.  If we deleted the events in update_server_on_end(), we'd be doing an O(N) search that provides no benefit to calendaring.  It would only slow it down.  Knowing that calendaring is one of the slowest areas of the code, I think we should leave this alone.